pkg/kubectl/cmd/attach.go

```diff
@@ -116,12 +122,37 @@ type AttachOptions struct {
 // Complete verifies command line arguments and loads data from the command environment
 func (p *AttachOptions) Complete(f cmdutil.Factory, cmd *cobra.Command, argsIn []string) error {
 	if len(argsIn) == 0 {
-		return cmdutil.UsageError(cmd, "POD is required for attach")
+		return cmdutil.UsageError(cmd, "at least one argument is required for attach")
+	}
+	if len(argsIn) > 2 {
+		return cmdutil.UsageError(cmd, fmt.Sprintf("expected fewer than three arguments: POD or TYPE/NAME or TYPE NAME, saw %d: %s", len(argsIn), argsIn))
 	}
-	if len(argsIn) > 1 {
-		return cmdutil.UsageError(cmd, fmt.Sprintf("expected a single argument: POD, saw %d: %s", len(argsIn), argsIn))
+
+	cmdNamespace, _, err := f.DefaultNamespace()
+	if err != nil {
+		return err
+	}
+
+	if strings.Contains(argsIn[0], "/") || len(argsIn) == 2 {
```

Comment:
You don't need to do this manual parsing. You should use ResourceNames in the resource builder instead. Have a look at how resource support is implemented in `kubectl logs` and model `kubectl attach` after it.

https://github.com/kubernetes/kubernetes/blob/748fb52cd5c4999ad19fda177657a1185b7f49fe/pkg/kubectl/cmd/logs.go#L195